src/Chirp.Web/Pages/TimelineModel.cshtml.cs

```diff
@@ -17,99 +18,117 @@ public TimelineModel(IAuthorRepository authorRepository, ICheepService service,
+        
+        string text = Request.Form["Text"].ToString();
+        if (text.Length > 160) text = text.Substring(0, 160);
+        
+        _service.StoreCheep( new Cheep {AuthorId = (int)authorId, Text=text, TimeStamp = DateTime.Now} );
         return RedirectToPage();
     }
 
-    public bool AuthorLikesCheep(string name, int cheepId)
+    public bool AuthorLikesCheep(int cheepId)
     {
-        var authors = _authorRepository.FindAuthorsByName(name);
-        if (!authors.Any()) return false; // Should never happen
-        return _likeRepository.LikeExists(authors.First().AuthorId, cheepId);
+        var authorId = GetUserId();
+        return authorId != null && _likeRepository.LikeExists((int)authorId, cheepId);
     }
 
     public int GetLikeCount(int cheepId)
     {
         return _likeRepository.FindLikeCountByCheepId(cheepId);
     }
 
-    public bool LikesOwnCheep(string authorName, int cheepId)
+    public bool LikesOwnCheep(int cheepId)
     {
-        var authors = _authorRepository.FindAuthorsByName(authorName);
-        if (!authors.Any()) return false;
-        return _likeRepository.LikesOwnCheep(authors.First().AuthorId, cheepId);
+        var authorId = GetUserId();
+        return authorId != null && _likeRepository.LikesOwnCheep((int)authorId, cheepId);
     }
 
     public IActionResult OnPostLike(int cheepId)
     {
-        if (!User.Identity.IsAuthenticated) return Page();
+        if (!IsUserAuthenticated()) return Page();
 
-        int authorId = _authorRepository.FindAuthorsByName(User.Identity.Name).First().AuthorId;
-        _likeRepository.LikeCheep(authorId, cheepId);
+        var authorId = GetUserId();
+        if (authorId == null) return Page();
+
+        _likeRepository.LikeCheep((int)authorId, cheepId);
         return RedirectToPage();
     }
 
     public IActionResult OnPostUnlike(int cheepId)
     {
-        if (!User.Identity.IsAuthenticated) return Page();
+        if (!IsUserAuthenticated()) return Page();
+
+        var authorId = GetUserId();
+        if (authorId == null) return Page();
 
-        int authorId = _authorRepository.FindAuthorsByName(User.Identity.Name).First().AuthorId;
-        _likeRepository.UnlikeCheep(authorId, cheepId);
+        _likeRepository.UnlikeCheep((int)authorId, cheepId);
         return RedirectToPage();
     }
 
-    public bool AuthorFollowsAuthor(string followerName, string followedName)
+    public bool UserFollowsAuthor(string followedName)
     {
-        var followers = _authorRepository.FindAuthorsByName(followerName);
-        if (!followers.Any()) return false; // Should never happen
-
-        var followed = _authorRepository.FindAuthorsByName(followedName);
-        if (!followed.Any()) return false; // Should never happen
-
-        return _followRepository.FollowExists(followers.First().AuthorId, followed.First().AuthorId);
+        var followerId = GetUserId();
+        var followedId = GetUserId(followedName);
```

Comment:
Hvorfor kalder vi dette to gange?

Reply:
den ene er user, den anden er den der bliver fulgt